src/BlockTypes/Checkout.php

```diff
@@ -179,18 +179,41 @@ protected function hydrate_customer_payment_methods( AssetDataRegistry $data_reg
 		remove_filter( 'woocommerce_payment_methods_list_item', [ $this, 'include_token_id_with_payment_methods' ], 10, 2 );
 	}
 
+	/**
+	 * Returns current notices and clears them.
+	 *
+	 * @return array $captured_notices Notices returned by `wc_get_notices`.
+	 */
+	protected function capture_and_clear_notices() {
+		// Save previous notices so we can return them in a `captured_notices`
+		// property to be used in the frontend.
+		$captured_notices = wc_get_notices();
+
+		// Prevent carried notices from being converted to exceptions when getting
+		// the Cart and Checkout endpoints.
+		wc_clear_notices();
+
+		return $captured_notices;
+	}
+
 	/**
 	 * Hydrate the checkout block with data from the API.
 	 *
 	 * @param AssetDataRegistry $data_registry Data registry instance.
 	 */
 	protected function hydrate_from_api( AssetDataRegistry $data_registry ) {
+		$captured_notices = $this->capture_and_clear_notices();
```

Comment:
Do we want to capture notices every time `hydrate_from_api` is called, or should this be protected?